acvm_backend_barretenberg/src/composer.rs

```diff
@@ -142,15 +141,14 @@ impl Composer for Barretenberg {
         unsafe {
             proof_size = barretenberg_sys::composer::create_proof_with_pk(
                 pippenger_ptr,
-                &g2_data,
+                g2_data,
                 &proving_key,
                 &cs_buf,
                 &witness_buf,
                 p_proof,
             );
         }
 
-        std::mem::forget(g2_data);
```

Comment:
Why can we remove this forget? Is g2_data not still being free'd inside bberg?

Reply:
I was just following clippy, but I think it is due to `g2_data` being of `&Vec<u8>` type so it's not needed 🤔

Reply:
That may be the case as we are only fetching a reference to the Vec inside the method rather than it being shared amongst the composer? As long as we don't have seg faults this is fine with me, I just was surprised

Reply:
We removed all forget calls 🎉